doc/source/dev/toolchain.rst

```diff
@@ -96,7 +96,19 @@ starting from version 1.7.0).
 
 To maintain compatibility with a large number of platforms & setups, especially
 where using the official wheels (or other distribution channels like Anaconda
-or conda-forge) is not possible, SciPy keeps compatibility with old compilers.
+or conda-forge) is not possible, SciPy tries to keep compatibility with older
+compilers, on platforms that have not yet reached their official end-of-life.
+
+As explained in more detail below, the current minimal compiler versions are:
+
+==========  ===========================  ===============================  ============================
+ Compiler    Default Platform (tested)    Secondary Platform (untested)    Minimal Version
+==========  ===========================  ===============================  ============================
+ GCC         Linux                        AIX, Alpine Linux, OSX           GCC 8.x
+ LLVM        OSX                          Linux, FreeBSD, Windows          LLVM 12.x
```

Comment:
This is too aggressive it looks like - conda-forge is at `clang 11.1.0`. Although I have some memory about version numbers being messed up, and `11` actually being `12` (or vice versa)?

Reply:
Conda-forge is at clang 13 actually. But I'd be happy to say that clang 10+ should work, but nothing below 12 is tested.

(Or we add a dedicated CI job to compile with old LLVM)

Reply:
> But I'd be happy to say that clang 10+ should work, but nothing below 12 is tested.

Let's do that. Clang 12.0 is only 1y 3m old, that's pretty recent.

> (Or we add a dedicated CI job to compile with old LLVM)

no need